lib/http/response.rb

```diff
@@ -125,6 +129,15 @@ def cookies
       end
     end
 
+    def chunked?
+      encoding = headers.get(Headers::TRANSFER_ENCODING)
+      case encoding
+        when String then encoding == CHUNKED
+        when Enumerable then encoding.include?("chunked")
+        else false
+      end
```

Comment:
Oh. Now I see... Well, according to RFC logic should be a bit more "complex"

``` ruby
encodings = @headers.get(Headers::TRANSFER_ENCODING)

if CHUNKED == encodings.last
  read_till_last_chunk
elsif encodings.include?(CHUNKED)
  read_till_server_closes_connection
else
  read_content_length_amount_of_byte_or_till_nothing_got_sent
end
```

Although, I think we can simplify that to:

``` ruby
if CHUNKED == @headers.get(Headers::TRANSFER_ENCODING).last
  read_till_last_chunked
else
  read_as_usual
end
```

Leading us towards:

``` ruby
def chunked?
  CHUNKED == @headers.get(Headers::TRANSFER_ENCODING).last
end
```